docs/contents/dev-docs/building-controller.md

```diff
@@ -0,0 +1,83 @@
+# Building an ACK service controller
+
+This document continues our contributor-focused discussion by explaining how to
+build or regenerate an ACK service controller.
+
+## Prerequisites
+
+You should have forked the `github.com/aws-controllers-k8s/code-generator`
+repository and `git clone`'d it locally when [setting up](../setup) your
+development environment,
+
+With the prerequisites out of the way, let's move on to the first step:
+building the code generator.
+
+## Build code generator
+
+Building an ACK service controller (or regenerating an existing one from a
+newer API model file) requires the `ack-generate` binary, which is the main
+code generator CLI tool.
+
+To build the latest `ack-generate` binary, execute the following command from
+the root directory of the `github.com/aws-controllers-k8s/code-generator`
+source repository:
+
+```
+make build-ack-generate
+```
+
+!!! note "One-off build"
+    You only have to do this once, overall. In other words: unless we change
+    something upstream in terms of the code generation process, this is
+    a one-off operation. Internally, the Makefile executes an `go build` here.
+
+Don't worry if you forget this step, the script in the next step will complain
+with a message along the line of `ERROR: Unable to find an ack-generate binary`
+and will give you another opportunity to rectify the situation.
+
+## Build an ACK service controller
+
+Now that we have the basic code generation step done we will create the
+respective ACK service controller and its supporting artifacts.
+
+So first you have to select a service that you want to build and test.
+You do that by setting the `SERVICE` environment variable. Let's say we want
+to test the S3 service (creating an S3 bucket), so we would execute the
+following:
+
+```
+export SERVICE=s3
+```
+
+Now we are in a position to generate the ACK service controller for the S3 API.
+
+```
+make build-controller SERVICE=$SERVICE
```

Comment:
Which `github.com/aws/aws-sdk-go` version does this refer while generating the service controller?

Both s3 controller go.mod file https://github.com/aws-controllers-k8s/s3-controller/blob/main/go.mod#L7
and code-generator go.mod file https://github.com/aws-controllers-k8s/code-generator/blob/main/go.mod#L6
have entry for `aws-sdk-go` and their values can differ.

Reply:
@kumargauravsharma You can set `aws-sdk-go` version now using `AWS_SDK_GO_VERSION` environment variable: https://github.com/aws-controllers-k8s/code-generator/commit/92458ababbeb0cd0f00b93e149ccc6b4e42f2835

Reply:
>Which github.com/aws/aws-sdk-go version does this refer while generating the service controller?

if `AWS_SDK_GO_VERSION` is not set, `ack-generate` will use the service controller `go.mod` file, if this file doesn't exist yet, it will try to find a `go.mod` file on your current working directory